app/authentication/user_id_generator.py

```diff
@@ -19,17 +22,15 @@ def generate_id(token):
 
         if ru_ref and collection_exercise_sid and eq_id:
             logger.debug("Using values %s, %s and %s with a salt", ru_ref, collection_exercise_sid, eq_id)
-            salt = settings.EQ_SERVER_SIDE_STORAGE_USER_ID_SALT
+            salt = to_bytes(settings.EQ_SERVER_SIDE_STORAGE_USER_ID_SALT)
+            user_id_material = ru_ref + collection_exercise_sid + eq_id
 
-            sha256 = hashlib.sha256()
-            sha256.update(to_str(ru_ref).encode('utf-8'))
-            sha256.update(to_str(collection_exercise_sid).encode('utf-8'))
-            sha256.update(to_str(eq_id).encode('utf-8'))
-            sha256.update(to_str(salt).encode('utf-8'))
-
-            user_id = sha256.hexdigest()
-            logger.debug("User ID is %s", user_id)
-            return user_id
+            kdf = PBKDF2HMAC(algorithm=hashes.SHA256(), length=32, salt=salt, iterations=100000, backend=backend)
```

Comment:
Should this be defaulting to 100,000 iterations? Also consider making it a setting and impose a minimum limit.